File: pizzayolo/src/main/java/pizzayolo/entity/Creneau.java
package pizzayolo.entity;

import java.time.LocalDate;
import java.time.LocalTime;
import java.util.Objects;

import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.ManyToMany;
import javax.persistence.SequenceGenerator;
import javax.persistence.Table;

@Entity
@Table(name = "creneau")
@SequenceGenerator(name = "seqCreneau", sequenceName = "seq_creneau", initialValue = 1, allocationSize = 1)
public class Creneau {

	@Id
	@GeneratedValue(strategy = GenerationType.SEQUENCE, generator = "seqCreneau")
	@Column(name="id")
	private Long id;
	private LocalDate date;//a quoi ca sert ???
	@Column(name="debut")
	private LocalTime debut;
	@Column(name="fin")
	private LocalTime fin;

	@ManyToMany(mappedBy = "creneaux")
	private Employee employe;

	public Creneau() {
	}

	public Long getId() {
		return id;
	}

	public void setId(Long id) {
		this.id = id;
	}

	public LocalDate getDate() {
		return date;
	}

	public void setDate(LocalDate date) {
		this.date = date;
	}

	public LocalTime getDebut() {
		return debut;
	}

	public void setDebut(LocalTime debut) {
		this.debut = debut;
	}

	public LocalTime getFin() {
		return fin;
	}

	public void setFin(LocalTime fin) {
		this.fin = fin;
	}

	public Employee getEmploye() {
		return employe;
	}

	public void setEmploye(Employee employe) {
		this.employe = employe;
	}

	@Override
	public int hashCode() {
		return Objects.hash(id);
	}

	@Override
	public boolean equals(Object obj) {
		if (this == obj)
			return true;
		if (obj == null)
			return false;
		if (getClass() != obj.getClass())
			return false;
		Creneau other = (Creneau) obj;
		return Objects.equals(id, other.id);
	}

}

File: pizzayoloangular/src/app/model/creneau.ts
import { Time } from '@angular/common';
import { Employe } from './employe';

export class Creneau {
  public constructor(
    private _date?: string,
    private _debut?: Time,
    private _fin?: Time,
    private _listeEmploye?: Employe[],
    private _employe?: string
  ) {}

  get date(): string | undefined {
    return this._date;
  }

  set date(value: string | undefined) {
    this._date = value;
  }

  get debut(): Time | undefined {
    return this._debut;
  }

  set debut(value: Time | undefined) {
    this._debut = value;
  }

  get fin(): Time | undefined {
    return this._fin;
  }

  set fin(value: Time | undefined) {
    this._fin = value;
  }

  get listeEmploye(): Array<Employe> | undefined {
    return this._listeEmploye;
  }

  set listeEmploye(value: Array<Employe> | undefined) {
    this._listeEmploye = value;
  }

  get employe(): string | undefined {
    return this._employe;
  }

  set employe(value: string | undefined) {
    this._employe = value;
  }
}
